packages/react-native/ReactAndroid/src/main/jni/react/fabric/StateWrapperImpl.h

```diff
@@ -29,6 +29,7 @@ class StateWrapperImpl : public jni::HybridClass<StateWrapperImpl> {
   jni::local_ref<ReadableNativeMap::jhybridobject> getStateDataImpl();
   void updateStateImpl(NativeMap* map);
   void setState(std::shared_ptr<const State> state);
+  const std::shared_ptr<const State> getState() const;
```

Comment:
The first const doesn't add any value
```suggestion
 std::shared_ptr<const State> getState() const;
```

Reply:
On a second thought, I think returning a const ref might be a better option here to avoid an unnecessary copy?

Reply:
Yes, that's fine too. Although you do have to make sure you don't hold on to the shared_ptr beyond the lifecycle of the StateWrapperImpl then.